### PR TITLE
fix #11246: add gcvote request to gc online search results

### LINKED ISSUE #11246: GCVote seems missing in our new search?
image

On the screenshots, it seems like the GCVote is missing on our new UI? Or was there a difference in the way the screenshots where created?

_Originally posted by @fm-sys in https://github.com/cgeo/cgeo/discussions/11069#discussioncomment-995496_

### PR DESCRIPTION
**Pull request by eddiemuc**
fix #11246: add gcvote request to gc online search results

### CODE REVIEW

File: main/src/cgeo/geocaching/connector/gc/GCMap.java [hunk 1]
```diff
@@ -148,7 +148,7 @@ public static SearchResult searchByFilter(@NonNull final GeocacheFilter filter,
             fillForBasicFilter(baseFilter, search);
         }
 
-        return GCWebAPI.searchCaches(search);
+        return GCWebAPI.searchCaches(search, true);
```
Comment (moving-bits) [inline main/src/cgeo/geocaching/connector/gc/GCMap.java:151]: shouldn't including GCVotes depend on the system setting for GCVote connector being activated?

Comment (Lineflyer) [reply main/src/cgeo/geocaching/connector/gc/GCMap.java:151]: Actually it was always like that:
In the GCVote settings the default is to load the ratings unless the user explicitly switches it off.
Autorizing to post votes is independant from that.

However that mentioned setting for loading votes should be respected.

Comment (moving-bits) [reply main/src/cgeo/geocaching/connector/gc/GCMap.java:151]: Yes, that's what I meant. If someone switches GCVote setting of, no http(s) call to GCVote should be made.

Comment (eddiemuc) [reply main/src/cgeo/geocaching/connector/gc/GCMap.java:151]: The switch in the settings is respected further down the call stack in the GCVote class itself

File: main/src/cgeo/geocaching/connector/gc/GCWebAPI.java [hunk 2]
```diff
@@ -811,6 +812,9 @@ static SearchResult searchCaches(final WebApiSearch search) {
         }
 
         result.addAndPutInCache(foundCaches);
+        if (includeGcVote) {
+            GCVote.loadRatings(foundCaches);
```
Comment (eddiemuc) [inline main/src/cgeo/geocaching/connector/gc/GCWebAPI.java:816]: `GCVote.loadRatings` checks for the Settings (whether to load GCVote or not).before issuing any requests to the server.

File: main/src/cgeo/geocaching/connector/gc/GCMap.java (hunk 1 above)
Comment (moving-bits) [reply main/src/cgeo/geocaching/connector/gc/GCMap.java:151]: ah, thanks for clarifying 👍

**Pull request merged**